File: BOJ/python3/18110.py
'''
solved.ac
https://www.acmicpc.net/problem/18110
'''
from collections import deque
import sys
import time
start = time.time()  # 시작 시간
sys.stdin = open("input.txt", "r")  # input.txt 파일로 입력


def get_avg(arr):
    arr.sort()
    arr = deque(arr)
    cnt = int(n * 0.15 + 0.5)
    for i in range(cnt):
        arr.popleft()
        arr.pop()

    sum_arr = 0
    for ele in arr:
        sum_arr += ele

    return int(sum_arr / len(arr) + 0.5)


scores = []

n = int(input())
for _ in range(n):
    score = int(sys.stdin.readline())
    scores.append(score)

if n == 0:
    print(0)
else:
    avg = get_avg(scores)
    print(avg)


# 현재시각 - 시작시간 = 실행 시간(sec)
print("--------------------")
print(f"Elapsed Time: {round(time.time() - start, 3)}s")

File: BOJ/nodejs/18110.js
/*
solved.ac
https://boj.kr/18110
 */
const start = new Date().getTime();

const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .split("\n");

const n = 1 * input[0];
const scores = new Array(n).fill(0);
for (let i = 0; i < n; i++) {
  scores[i] = 1 * input[1 + i];
}

scores.sort((a, b) => a - b);
const cnt = Math.round(n * 0.15);
const slicedScores = scores.slice(cnt, n - cnt);
const ans = !n
  ? 0
  : Math.round(
      slicedScores.reduce((acc, cur) => acc + cur, 0) / slicedScores.length
    );

console.log(ans);

const end = new Date().getTime();
console.log("--------------------");
console.log(`Elapsed Time: ${(new Date().getTime() - start) / 1000}s`);
